src/interceptors/response.ts

```diff
@@ -297,7 +303,11 @@ export function defaultResponseInterceptor(axios: AxiosCacheInstance): ResponseI
       }
 
       // Rejects all other requests waiting for this response
-      await rejectResponse(id, config);
+      await rejectResponse(
```

Comment:
the specific `if` condition where this is located isnt ideal, (_"This will only not be loading if the interceptor broke"_ at line 293), also, what will happen on other `rejectResponse` calls? Shouldn't they also keep cache on `ERR_CANCELED`?

Reply:
If I understand the code correctly, we go there when the interceptor is broken, because of the cancel. Because we're always pre request here, since the request is cached but we cancel before the a-c-i provide the cache results.

Reply:
What I'm asking is like: Cancellation could happen in 2 places if Im not wrong:

1. when requesting
2. while receiving the reply.

In your test, only the opt 1 is being tested, and that option ends up in this if condition, but what about the second option, I don't know, but don't think it will be in this same if as well.

Reply:
What I'm asking is like: Cancellation could happen in 2 places if Im not wrong:

1. when requesting
2. while receiving the reply.

In your test, only the opt 1 is being tested, and that option ends up in this if condition, but what about the second option, I don't know, but don't think it will be in this same if as well.

Reply:
I don't know how to test 2. because I don't know how to make a test dispatched to the network without putting a real url, but since you didn't do any of this in the current code base I'm not sure if I can introduce this kind of test.

Reply:
I modified the test, to make the second request wait for a microtask which should allow axios to make the request. But it doesn't make much sense because in this case, since there's a cache, it's not going to make a request anyway. At least the test exist.